keras_cv/metrics/coco/base_test.py

```diff
@@ -115,7 +113,37 @@ def test_matches_single_false_positive(self):
         )
         metric.update_state(y_true, y_pred)
 
-        # shape = [1, 1, 1, 1]
         self.assertEqual(0.0, metric.true_positives[0, 0].numpy())
-        # shape = [1, 1, 1, 1]
         self.assertEqual(1.0, metric.false_positives[0, 0].numpy())
+
+    def test_merge_state(self):
+        y_true = tf.constant([[[0, 0, 100, 100, 1]]], dtype=tf.float32)
+        y_pred = tf.constant([[[0, 50, 100, 150, 1, 1.0]]], dtype=tf.float32)
+
+        m1 = COCOBase(
+            iou_thresholds=[0.95],
+            category_ids=[1],
+            area_range=(0, 100000**2),
+            max_detections=1,
+        )
+        m2 = COCOBase(
+            iou_thresholds=[0.95],
+            category_ids=[1],
+            area_range=(0, 100000**2),
+            max_detections=1,
+        )
+
+        m1.update_state(y_true, y_pred)
+        m1.update_state(y_true, y_pred)
+        m2.update_state(y_true, y_pred)
+
+        metric_result = COCOBase(
+            iou_thresholds=[0.95],
+            category_ids=[1],
+            area_range=(0, 100000**2),
+            max_detections=1,
+        )
+        metric_result.merge_state([m1, m2])
+
+        self.assertEqual(0.0, metric_result.true_positives[0, 0].numpy())
```

Comment:
if the weights shape is [1, 1], then let's change the assert here to be [[0.0]] and metric_result.true_positives